user-console/src/main/java/org/pentaho/mantle/client/admin/EmailAdminPanel.java

```diff
@@ -204,4 +331,48 @@ protected boolean isPortValid( String portValue ) {
       return false;
     }
   }
+
+  protected boolean isSmtpMicrosoftOAuth( String protocol, String smtpHost, String authMechanism ) {
+    return ( isSmtpProtocolFamily( protocol ) && ( smtpHost.contains( "outlook" ) || smtpHost.contains( "office" ) ) && authMechanism.equals( IEmailConfiguration.AUTH_TYPE_XOAUTH2 ) );
+  }
+
+  protected boolean isSmtpProtocolFamily( String protocol ) {
+    return ( protocol.equals( IEmailConfiguration.PROTOCOL_SMTP ) ) || ( protocol.equals( IEmailConfiguration.PROTOCOL_SMTPS ) );
+  }
+
+  protected void selectProtocol( String protocol ) {
+    protocolsListBox.setSelectedIndex( -1 );
+    if ( !StringUtils.isEmpty( protocol ) ) {
+      for ( int i = 0; i < protocolsListBox.getItemCount(); ++i ) {
+        if ( protocol.equalsIgnoreCase( protocolsListBox.getValue( i ) ) ) {
+          protocolsListBox.setSelectedIndex( i );
+          break;
+        }
+      }
+    }
+  }
+
+  protected void selectAuthMechanism( String authMechanism ) {
+    authenticationTypeLB.setSelectedIndex( -1 );
+    if ( !StringUtils.isEmpty( authMechanism ) ) {
+      for ( int i = 0; i < authenticationTypeLB.getItemCount(); ++i ) {
+        if ( authMechanism.equalsIgnoreCase( authenticationTypeLB.getValue( i ) ) ) {
```

Comment:
add check for null for Listbox here. For now it unlikely, but in later releases things could change. And we get bugs all the time due to not checking for null